tuxemon/core/monster.py

```diff
@@ -432,6 +444,47 @@ def get_evolution(self, path):
                     return evolution['monster_slug']
         return None
 
+    def get_sprite(self, sprite, **kwargs):
+        """Gets a specific type of sprite for the monster.
+
+        :rtype: Pygame surface
+        :returns: The surface of the monster sprite
+        """
+        if sprite == "front":
+            surface = tools.load_sprite(self.front_battle_sprite, **kwargs)
+        elif sprite == "back":
+            surface = tools.load_sprite(self.back_battle_sprite, **kwargs)
+        elif sprite == "menu":
+            surface = tools.load_sprite(self.menu_sprite, **kwargs)
+        else:
+            raise ValueError("Cannot find sprite for: {}".format(sprite))
+
+        # Apply flairs to the monster sprite
+        for flair in self.flairs:
+            print(flair.name)
+            flair_path = self.get_sprite_path("gfx/sprites/battle/{}-{}-{}".format(self.slug, sprite, flair.name))
+            if flair_path != MISSING_IMAGE:
+                flair_sprite = tools.load_sprite(flair_path, **kwargs)
+                surface.image.blit(flair_sprite.image, (0, 0))
+
+        return surface
+
+    def set_flairs(self):
+        """Sets the flairs of this monster if they were not already configured
+
+        :rtype: None
+        :returns: None
+        """
+        if len(self.flairs) > 0 or self.slug == "":
+            return
+
+        results = monsters.lookup(self.slug)
+        flairs = results.get("flairs")
+        if flairs:
```

Comment:
```python

if flairs:
  flair_info = random.choice(flairs)
  flair = Flair(**flair_info)
  self.flairs[flair.category] = flair
```

Reply:
I think that format would break functionality. The flair categories are intended to be all set, only the values are meant to be random. It looks like it's picking a random flair group instead.

Reply:
```python
# move to top of file
from itertools import chain

if flairs:
  all_flairs = list(chain.from_iterable(i.values() for i in flairs))
  flair_info = random.choice(all_flairs)
  flair = Flair(**flair_info)
  self.flairs[flair.category] = flair

Reply:
`TypeError: type object argument after ** must be a mapping, not unicode`

Reply:
```python
category = random.choice(flairs)
name = random.choice(flairs[category])
flair = Flair(category, name)
```

Reply:
Also crashed: `TypeError: list indices must be integers, not dict`. It also picked a random flair to set instead of setting all of them, only the value is random but all flairs must be set. I managed to get it working perfectly with this:

```
for flair in flairs:
    flair = Flair(flair['name'], random.choice(flair['values']))
    self.flairs[flair.category] = flair
```